frontend/src/hooks/useIntervalRange.ts

```diff
@@ -0,0 +1,108 @@
+import getLocalStorageKey from 'api/browser/localstorage/get';
+import dayjs from 'dayjs';
+import getTimeString from 'lib/getTimeString';
+import { useCallback } from 'react';
+import { useLocation } from 'react-router-dom';
+import { getDiffs } from 'utils/getDiffs';
+
+import ROUTES from '../constants/routes';
+import {
+	getDefaultOption,
+	Time,
+} from '../container/TopNav/DateTimeSelection/config';
+
+interface UseIntervalRangeI {
+	getTime: () => [number, number] | undefined;
+	getCustomOrIntervalTime: (time: Time, currentRoute: string) => Time;
+}
+
+export function useIntervalRange(): UseIntervalRangeI {
+	const { search, pathname } = useLocation();
+	const params = new URLSearchParams(search);
+	const searchStartTime = params.get('startTime');
+	const searchEndTime = params.get('endTime');
+
+	const localstorageStartTime = getLocalStorageKey('startTime');
+	const localstorageEndTime = getLocalStorageKey('endTime');
+	const getQueryInterval = useCallback(
+		(searchStartTime: string) => {
+			const lastRefresh = dayjs(
+				new Date(parseInt(getTimeString(searchStartTime), 10)),
+			);
+			const { minutedDiff, hoursDiff, daysDiff } = getDiffs(lastRefresh);
+
+			if (daysDiff > 1) {
+				return '1week';
+			}
+			if (hoursDiff > 6) {
+				return '1day';
+			}
+			if (hoursDiff > 1) {
+				return '6hr';
+			}
+			if (hoursDiff > 0) {
+				return '1hr';
+			}
+			if (minutedDiff > 15) {
+				return '30min';
+			}
+			if (minutedDiff > 5) {
+				return '15min';
+			}
+			if (minutedDiff > 1) {
+				return '5min';
+			}
+
+			return ROUTES.SERVICE_MAP === pathname ? '1week' : 'custom';
```

Comment:
any reason for `ROUTES.SERVICE_MAP` ?

Reply:
![image](https://user-images.githubusercontent.com/50579516/218398616-dc9f7c3a-3a78-4a94-bdab-92800096f3b0.png)
I could not see the custom option inside the select on this page

Reply:
@KolesnikAntony got it